src/braket/default_simulator/openqasm/interpreter.py

```diff
@@ -577,8 +577,8 @@ def _(self, node: FunctionCall) -> Optional[QASMNode]:
         self._uses_advanced_language_features = True
         function_name = node.name.name
         arguments = self.visit(node.arguments)
-        if function_name in builtin_functions:
-            return builtin_functions[function_name](*arguments)
+        if function_name in dir(BuiltinFunctions) and not function_name.startswith("__"):
+            return getattr(BuiltinFunctions, function_name)(*arguments)
```

Comment:
I could keep the dict, but I'm not convinced it adds any specific value here. If you feel like it's a better setup, then I'm happy to bring it back though

Reply:
what if I moved the BuiltinFunctions to a separate module and replaced the dict with a function `get_builtin_method(name)`. That way the interface wouldn't be meta-programming, rather pretty analogous to a dict look-up, but we could still simplify the implementation with getattr under the hood to avoid manually building a redundant map

Reply:
Yeah, definitely like that more!